packages/blaze-tx/src/tx.ts

```diff
@@ -1568,56 +1568,62 @@ export class TxBuilder {
         spareInputs.push(utxo);
       }
     }
-    // Perform coin selection to cover any negative excess value.
-    const selectionResult = this.coinSelector(
-      spareInputs,
-      value.negate(value.negatives(excessValue)),
-    );
-    // Update the excess value and spare inputs based on the selection result.
-    excessValue = value.merge(excessValue, selectionResult.selectedValue);
-    spareInputs = selectionResult.inputs;
-    // Add selected inputs to the transaction.
-    if (selectionResult.selectedInputs.length > 0) {
-      for (const input of selectionResult.selectedInputs) {
-        this.addInput(input);
-      }
-    } else {
-      if (this.body.inputs().size() == 0) {
-        if (!spareInputs[0]) {
-          throw new Error(
-            "No spare inputs available to add to the transaction",
-          );
+
+    if (coinSelection) {
+      // Perform coin selection to cover any negative excess value.
+      const selectionResult = this.coinSelector(
+        spareInputs,
+        value.negate(value.negatives(excessValue)),
+      );
+      // Update the excess value and spare inputs based on the selection result.
+      excessValue = value.merge(excessValue, selectionResult.selectedValue);
+      spareInputs = selectionResult.inputs;
+      // Add selected inputs to the transaction.
+      if (selectionResult.selectedInputs.length > 0) {
+        for (const input of selectionResult.selectedInputs) {
+          this.addInput(input);
         }
-        // Select the input with the least number of different multiassets from spareInputs
-        const [inputWithLeastMultiAssets] = spareInputs.reduce(
-          ([minInput, minMultiAssetCount], currentInput) => {
-            const currentMultiAssetCount = value.assetTypes(
-              currentInput.output().amount(),
+      } else {
+        if (this.body.inputs().size() == 0) {
+          if (!spareInputs[0]) {
```

Comment:
hmm; if coin selection is disabled, this looks like we're doing some weird backup coin selection?

I'd expect, if you disable coin selection, that we just check if there's enough assets on the already selected inputs and fail otherwise.

Reply:
I think this defeats the purpose of the opt out feature though; the reason someone would want to opt out is because they want complete control over which inputs are included.

Reply:
No because this is only happening if the coin selection doesn't find anything. i.e. it is only running if coin selection is set to true.

Reply:
? that doesn't make sense.

It's running if coin selection is falsy, which I thought meant someone had opted out.

Reply:
(for posterity, I was wrong here :sweat_smile: I was misreading the indentation of the else block)